libraries/AP_HAL/utility/Socket.cpp

```diff
@@ -327,7 +347,9 @@ ssize_t SOCKET_CLASS_NAME::recv(void *buf, size_t size, uint32_t timeout_ms)
     socklen_t len = sizeof(struct sockaddr_in);
     int fin = get_read_fd();
     ssize_t ret;
-    ret = CALL_PREFIX(recvfrom)(fin, buf, size, MSG_DONTWAIT, (sockaddr *)&last_in_addr[0], &len);
+    uint32_t in_addr[4] = {};
```

Comment:
Because we need to pass in an empty structure, so it receives from any client and it gets updated with the latest packets source. I think I need to memcpy only after successful receive.

Reply:
fixed